### PR TITLE
#36899 Fill in Document metadata on process link

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/CreateDocumentRequest.kt [hunk 1]
```diff
@@ -26,12 +26,14 @@ class CreateDocumentRequest(
     @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
     val creatiedatum: LocalDate = LocalDate.now(),
     val titel: String,
+    val vertrouwelijkheidaanduiding: String? = null,
```
[the remark this answers is not in the document]
Comment (Klaas-Ritense) [reply zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/CreateDocumentRequest.kt:29]: These fields are required in a different place; the plugin properties. But here I made them optional. 

The reason is that these fields are not required in the document api. And I wanted to make this request as similar to the documentation as possible: https://openzaak-zgw.test.denhaag.nl/documenten/api/v1/schema/#operation/enkelvoudiginformatieobject_list

File: zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/ConfidentialityNotice.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    INTERN("intern"),
+    ZAAKVERTROUWELIJK("zaakvertrouwelijk"),
+    VERTROUWELIJK("vertrouwelijk"),
+    CONFIDENTIEEL("confidentieel"),
+    GEHEIM("geheim"),
+    ZEER_GEHEIM("zeer_geheim");
+
+    companion object {
+        fun fromKey(key: String): ConfidentialityNotice {
+            for (confidentialityNotice in ConfidentialityNotice.values()) {
```
[the remark this answers is not in the document]
Comment (Klaas-Ritense) [reply zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/ConfidentialityNotice.kt]: changed

File: zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/CreateDocumentRequest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (theo-ritense) [reply zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/CreateDocumentRequest.kt:29]: Let's discuss next week with the team. I feel like it should be required to improve data completeness and have it match the plugin (action) properties.

[the remark this answers is not in the document]
Comment (marijnritense) [reply zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/CreateDocumentRequest.kt:29]: I think I agree with Klaas, this in particular is a match with what is described in the documentation, which supports multiple use cases. The plugin action is meant for a specific use case, in which we want to make every argument required.

**Review by Rick-Ritense — APPROVED** [review]

**Pull request merged**